lua/legendary/data/function.lua

```diff
@@ -26,6 +47,29 @@ function Function:parse(tbl) -- luacheck: no unused
   instance.opts = tbl.opts or {}
   instance:parse_filters(tbl.filters)
 
+  -- By default, function can be run in all modes, so mode_mapping is empty
+  instance.mode_mappings = vim.tbl_islist(tbl.mode) and tbl.mode or {}
+
+  -- If tbl = { fn, mode = "n" }
+  if type(tbl.mode) == 'string' then
+    table.insert(instance.mode_mappings, tbl.mode)
+  end
+  -- only tbl = { fn, mode = { 'n' } }
+  -- Reassing implementation with a current-mode hanlder
+  if not vim.tbl_isempty(instance.mode_mappings) then
+    local impl = instance.implementation
+    instance.implementation = function()
+      local modeCurrent = vim.fn.mode()
+      for _, modeInstance in ipairs(instance.mode_mappings) do
+        if modeCurrent == modeInstance then
+          impl()
+        end
+      end
+    end
+  end
+  -- mode_mapping is going to remain static during legendary runtime
+  -- so we can cache current mapping state
+  instance._mode_switched = vim.tbl_islist(instance.mode_mappings) and not vim.tbl_isempty(instance.mode_mappings)
```

Comment:
It doesn't feel necessary to me to cache this.

```suggestion
```

Reply:
It's used for displaying: https://github.com/mrjones2014/legendary.nvim/pull/415/commits/29513d11cf20d125ad9a26645ab7d31b43db39da#r1366849251 

; It's essential for both performance reason and readability

Reply:
I really don't think it will have a noticeable impact on performance, and I don't see how it affects readability either, personally.

If you can show some benchmarks proving otherwise I'm all ears.

Reply:
@mrjones2014 It's cheap to check once and leave, then checking whether the list is empty all the time and it doesn't check. This is obvious.

Reply:
It’s an extremely inexpensive check. It doesn’t need to be cached.